### PR TITLE
Add support for parameter trusted MODULES-1658

### PR DESCRIPTION
**Pull request by mkrakowitzer**
* Add support for paramater trusted, valid options are 'true' and false. defaults to false. True sets the value to trusted=yes.

trusted=yes can be set to indicate that packages from this source are always authenticated even if the Release file is not signed or the signature can't be checked.

* Update documentation

### CODE REVIEW

**Comment by mkrakowitzer** [comment]
I think the master branch is currently broken, which is why my travis CI build is failing :(

**Comment by WolverineFan** [comment]
Yes master is broken (though it seems to be due to a CI environment change rather than anything committed to master).  There are 2 pull requests in the queue that fix the problem.  Once one of those gets merged you can rebase your change on top of latest master.

**Comment by daenney** [comment]
I understand the reason for wanting this but I hate that apt supports this faking of authenticity. It seems the CI problems where resolved. Could you rebase and try this again?

**Comment by mkrakowitzer** [comment]
@daenney ok, rebased.

Im not in favour of it either, its a long story why I need this which I wont bore you with.

File: spec/defines/source_spec.rb [hunk 1]
```diff
@@ -55,6 +55,7 @@
         'key_source'        => 'http://apt.puppetlabs.com/pubkey.gpg',
         'pin'               => '10',
         'architecture'      => 'x86_64',
+        'trusted'           => true,
```
Comment (daenney) [inline spec/defines/source_spec.rb]: Why are we modifying an existing test?

Comment (mkrakowitzer) [inline spec/defines/source_spec.rb]: The test is called 'no defaults', if I left it then it would be a default.

Comment (daenney) [inline spec/defines/source_spec.rb]: Oh right, I missed that part.

**Comment by daenney** [comment]
@mhaskel @cyberious Comments?

**Comment by cyberious** [comment]
@daenney looks sane to me

**Comment by daenney** [comment]
Thank you for your contribution! 'till next time!

**Pull request merged**